app/javascript/serviceworker/store.ts

```diff
@@ -1,58 +1,78 @@
 import { openDB, DBSchema } from "idb";
 
+type StoreName = "delayedRequests";
+
 interface RequestObject {
   id?: number;
   url: string;
-  data: any;
+  body: any;
 }
 
 interface OfflineDatabase extends DBSchema {
-  requests: {
+  delayedRequests: {
     key: number;
     value: RequestObject;
+    indexes: { url: string };
+  };
+  cachedResponses: {
+    key: number;
+    value: RequestObject;
+    indexes: { url: string };
   };
-}
-
-interface Db {
-  createObjectStore: (
-    arg0: string,
-    arg1: { keyPath: string; autoIncrement: boolean }
-  ) => void;
 }
 
 const DB_NAME = "offline";
 const DB_VERSION = 1;
```

Comment:
If we're upgrading the DB should we also increment the version here?

Reply:
Oh, I see, I was confused ... we aren't upgrading, we are just changing how we _would_ upgrade. 🤦